diff --git a/vendor/go.podman.io/storage/drivers/overlay/overlay.go b/vendor/go.podman.io/storage/drivers/overlay/overlay.go
index 223181cf0..362c5ec5a 100644
--- a/vendor/go.podman.io/storage/drivers/overlay/overlay.go
+++ b/vendor/go.podman.io/storage/drivers/overlay/overlay.go
@@ -1545,6 +1545,7 @@ func (d *Driver) get(id string, disableShifting bool, options graphdriver.MountO
 	if !d.SupportsShifting(options.UidMaps, options.GidMaps) || options.DisableShifting {
 		disableShifting = true
 	}
+	logrus.Debugf("disableShifting: %s", strconv.FormatBool(disableShifting))
 
 	logLevel := logrus.WarnLevel
 	if unshare.IsRootless() {
@@ -2685,6 +2686,9 @@ func (d *Driver) checkAndRecordIDMappedSupport() bool {
 func (d *Driver) SupportsShifting(uidmap, gidmap []idtools.IDMap) bool {
 	if os.Getenv("_CONTAINERS_OVERLAY_DISABLE_IDMAP") == "yes" {
 		return false
 	}
+	if _, ok := os.LookupEnv("_CONTAINERS_FORCE_SHIFTING"); ok {
+		return true
+	}
 	if d.options.mountProgram != "" {
 		// fuse-overlayfs supports only contiguous mappings, since it performs the mapping on the
 		// upper layer too, to avoid https://github.com/containers/podman/issues/10272
